tools/autograd/templates/Functions.cpp

```diff
@@ -531,7 +532,7 @@ Tensor select_equals_backward(Tensor grad, const Tensor & input, const Tensor &
 }
 
 Tensor index_select_backward(Tensor grad, int64_t dim, Tensor indices, IntList sizes, bool keepdim) {
-  if (!keepdim) {
+  if (!keepdim && sizes.size() > 0) {
```

Comment:
Is there a test case that covers this change?

Reply:
It comes up when you turn on 0-dim tensors in TH (see https://github.com/pytorch/pytorch/pull/10383).